_posts/en/newsletters/2024-05-31-newsletter.md

```diff
@@ -0,0 +1,237 @@
+  can provide much of the same features at a similar cost should be
+  considered mutually exclusive, or whether it would make sense to
+  activate multiple proposals and let developers use whichever
+  alternative they prefer.
+
+  Anthony Towns [replies][towns exclusive] to multiple points, including
+  suggesting that overlapping features by themselves are not a problem
+  but features that aren't used because everyone prefers an alternative
+  may produce several problems.  He suggests anyone who favors a
+  particular proposal test using its features in pre-production software
+  to get a feel for them, especially in comparison to alternative ways
+  the feature can be added to Bitcoin.
+
+## Selected Q&A from Bitcoin Stack Exchange
+
+*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
+contributors look for answers to their questions---or when we have a
+few spare moments to help curious or confused users.  In
+this monthly feature, we highlight some of the top-voted questions and
+answers posted since our last update.*
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aa
+nswer -->{% endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
+
+- [What's the smallest possible coinbase transaction / block size?]({{bse}}122951)
+  Antoine Poinsot explains minimum restrictions around the coinbase transaction
+  and concludes the smallest possible valid Bitcoin block at current block heights is 145 bytes.
+
+- [Understanding Script's number encoding, CScriptNum]({{bse}}122939)
+  Antoine Poinsot describes how CScriptNum represents integers in Bitcoin's
+  Script, provides some example encodings, and links to two serialization
+  implementations.
+
+- [Is there a way to make a BTC wallet address public but hide how many BTC it contains?]({{bse}}122786)
+  Vojtěch Strnad points out that [silent payment][topic silent payments] reusable
+  payment addresses allow a public payment identifier to be posted without
+  observers being able to associate transactions paying to it.
+
+- [Testing increased feerates in regtest]({{bse}}122837)
+  In regtest, Ava Chow recommends using Bitcoin Core's test framework and
+  setting `-maxmempool` to a low value and `-datacarriersize` to a high value in
+  order to help simulate high feerate environments.
+
+- [Why is my P2P_V2 peer connected over a v1 connection?]({{bse}}122774)
+  Pieter Wuille hypothesizes that outdated peer addr information was what caused
+  a user to see a peer that supported BIP324 [encrypted transport][topic v2 p2p
+  transport] that was connected with a v1 non-encrypted connection.
+
+- [Does a P2PKH transaction send to the hash of the uncompressed key or the compressed key?]({{bse}}122875)
+  Pieter Wuille notes that both compressed and uncompressed public keys can be
+  used resulting in distinct P2PKH addresses, adding that P2PKH only supports
```

Comment:
```suggestion
  used resulting in distinct P2PKH addresses, adding that P2WPKH only supports
```

Reply:
Also, P2WPKH does support uncompressed keys, they're just non-standard by Bitcoin Core's policy. How about:
> P2WPKH only allows compressed public keys by policy and P2TR uses x-only public keys.

Reply:
Taking this suggestion, thanks @jirijakes and @vostrnad